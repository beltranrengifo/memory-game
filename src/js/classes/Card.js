export default class Card {
  constructor (name, clickHandler) {
    this.name = name
    this.clickHandler = clickHandler
    this.domElement = null
  }

  generateCard () {
    this.domElement = document.createElement('div')
    this.domElement.classList.add('card')
    this.domElement.setAttribute('brand', this.name)
    const cardElWrap = document.createElement('div')
    cardElWrap.classList.add('card__wrap')
    const cardElBrand = document.createElement('i')
    cardElBrand.classList.add('card__brand', 'fab', `fa-${this.name}`)
    cardElWrap.appendChild(cardElBrand)
    this.domElement.appendChild(cardElWrap)
    this.setListener()
    return this.domElement
  }

  setListener () {
    this.domElement.addEventListener('click', () => this.clickHandler(this.name), false)
  }
}
