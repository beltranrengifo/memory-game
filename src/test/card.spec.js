import Card from '@/js/classes/Card'

describe('Card', () => {
  let card
  let cardElement

  beforeEach(() => {
    card = new Card('twitter', jest.fn)
    cardElement = card.generateCard()
  })

  test('Card function', () => {
    expect(typeof Card).toBe('function')
  })

  test('Card name to be defined', () => {
    expect(card.name).toBeDefined()
  })

  test('Card DOM element to be defined', () => {
    expect(card.domElement).toBeDefined()
  })

  test('Card DOM element to be truthy', () => {
    expect(card.domElement).toBeTruthy()
  })

  test('Card handler to be defined', () => {
    expect(card.clickHandler).toBeDefined()
  })

  test('Card handler to be a function', () => {
    expect(typeof card.clickHandler).toBe('function')
  })

  test('Card generates DOM element and matches', () => {
    expect(cardElement).toMatchSnapshot()
  })

  test('spyOn card setListener', () => {
    const spyOnSetListener = jest.spyOn(card, 'setListener')
    card.setListener()
    expect(spyOnSetListener).toHaveBeenCalled()
  })

  test('spyOn card generateCard', () => {
    const spyOnGenerateCard = jest.spyOn(card, 'generateCard')
    card.generateCard()
    expect(spyOnGenerateCard).toHaveBeenCalled()
  })

  test('spyOn card clickHandler', () => {
    const spyOnClickHandler = jest.spyOn(card, 'clickHandler')
    card.domElement.click()
    expect(spyOnClickHandler).toHaveBeenCalled()
  })
})
