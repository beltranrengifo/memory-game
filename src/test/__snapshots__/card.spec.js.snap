// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card generates DOM element and matches 1`] = `
<div
  brand="twitter"
  class="card"
>
  <div
    class="card__wrap"
  >
    <i
      class="card__wrap fab fa-twitter"
    />
  </div>
</div>
`;
